app/models/manageiq/providers/ansible_runner_workflow.rb

```diff
@@ -55,7 +55,7 @@ def execute
   def poll_runner
     response = Ansible::Runner::ResponseAsync.load(context[:ansible_runner_response])
     if response.running?
-      if started_on + options[:timeout] < Time.now.utc
+      if options[:timeout].present? && (started_on + options[:timeout]) < Time.now.utc
         response.stop
 
         queue_signal(:abort, "ansible #{execution_type} has been running longer than timeout", "error")
```

Comment:
@lfu in the non-automate case where timeout *is* passed, then this will try to abort after doing `response.stop` and then I think it will hit the process_abort method and try to response.stop again.  If so, I think you need to remove the response.stop 2 lines up and just let it flwo through your new abort method.

Reply:
I'm sure `queue_signal(:abort)` works!